store/buildSkyblockCalendar.js

```diff
@@ -0,0 +1,256 @@
+
+const zooStart = yearZero + yearMs * 66;
+const zooTimeLength = yearMs / 2;
+
+const pets = [
+  'ELEPHANT',
+  'GIRAFFE',
+  'BLUE_WHALE',
+  'TIGER',
+  'LION',
+  'MONKEY',
+];
+
+function getOffset(month, day) {
+  return months.indexOf(month) * monthLength * dayMs + (day - 1) * dayMs;
+}
+
+function getZooPet(time) {
+  const iterations = Math.floor((time - zooStart) / zooTimeLength);
+
+  return pets[iterations % pets.length];
+}
+
+function getJacobEventTimes() {
+  const times = [];
+
+  for (let month = 0; month < 12; month++) {
+    let day = 2;
+    while (day <= 31) {
+      times.push({
+        start: getOffset(months[month], day),
+        end: getOffset(months[month], day),
+      });
+
+      if (day === 30) break;
+      day += 3;
+      if (day > 31) {
+        day %= 31;
+        month++;
+      }
+    }
+  }
+  return times;
+}
+
+const events = [
+  {
+    name: 'Bank Interest',
+    times: [
+      {
+        start: getOffset('Early Spring', 1),
+        end: getOffset('Early Spring', 1),
+      },
+      {
+        start: getOffset('Early Summer', 1),
+        end: getOffset('Early Summer', 1),
+      },
+      {
+        start: getOffset('Early Autumn', 1),
+        end: getOffset('Early Autumn', 1),
+      },
+      {
+        start: getOffset('Early Winter', 1),
+        end: getOffset('Early Winter', 1),
+      },
+    ],
+  },
+  {
+    name: 'Election Over',
+    times: [
+      {
+        start: getOffset('Late Spring', 27),
+        end: getOffset('Late Spring', 27),
+      },
+    ],
+  },
+  {
+    name: 'Traveling Zoo',
+    times: [
+      {
+        start: getOffset('Early Summer', 1),
+        end: getOffset('Early Summer', 3),
+      },
+      {
+        start: getOffset('Early Winter', 1),
+        end: getOffset('Early Winter', 3),
+      },
+    ],
+  },
+  {
+    name: 'Election Booth Opens',
+    times: [
+      {
+        start: getOffset('Late Summer', 27),
+        end: getOffset('Late Summer', 27),
+      },
+    ],
+  },
+  {
+    name: 'Fear Mongerer',
+    times: [
+      {
+        start: getOffset('Autumn', 26),
+        end: getOffset('Late Autumn', 3),
+      },
+    ],
+  },
+  {
+    name: 'Spooky Festival',
+    times: [
+      {
+        start: getOffset('Autumn', 29),
+        end: getOffset('Autumn', 31),
+      },
+    ],
+  },
+  {
+    name: 'Jerry\'s Workshop',
+    times: [
+      {
+        start: getOffset('Late Winter', 1),
+        end: getOffset('Late Winter', 31),
+      },
+    ],
+  },
+  {
+    name: 'Season of Jerry',
+    times: [
+      {
+        start: getOffset('Late Winter', 24),
+        end: getOffset('Late Winter', 26),
+      },
+    ],
+  },
+  {
+    name: 'New Year Celebration',
+    times: [
+      {
+        start: getOffset('Late Winter', 29),
+        end: getOffset('Late Winter', 31),
+      },
+    ],
+  },
+  {
+    name: 'Jacob\'s Farming Contest',
+    times: getJacobEventTimes(),
+  },
+];
+
+function buildSkyblockCalendar(event, years) {
```

Comment:
It would probably be beneficial to accept multiple events, separated by commas

Reply:
ok yeah i was thinking that would be a good idea